### DISCUSSION TITLE
Excluding compilation unit for referenced project

### DISCUSSION THREAD

**Discussion by birojnayak** [dotnet/roslyn#66745] [General]
If Project A is dependent on Project B , if I have already generated the compilation unit for Project B, can I exclude                 Compilation unit for Project B in Project A to save space ? If yes, how ?

**Comment by CyrusNajmabadi** [comment]
> to save space

What space are you referring to here?

**Reply by birojnayak** [reply]
memory...

**Reply by CyrusNajmabadi** [reply]
You can emit the one compilation into metadata, then reference that metadata.  That will allow the first compilation to get GC'ed.  However, three Compiler will then have to create an the symbols it needs from the metadata, instead of being able to directly use those from the direct compilation reference. 

Up to you if the trade off is worth it.

**Reply by birojnayak** [reply]
@CyrusNajmabadi thanks for the info.... so I dug deeper, the problem seems to be with unmanaged memory that's created via metadatareference/assemblymetadata/pemodule. Is there a way to get the assemblymetadata directly from the compilation unit, so that I explicitly call dispose, instead of waiting when gc would get to that ?
